code/datums/disease.dm

```diff
@@ -205,3 +205,157 @@ var/list/diseases = subtypesof(/datum/disease)
 	active_diseases -= src
 	. = ..()
 	return TA_PURGE_ME_NOW //immediate deletion
+
+
+/mob/living/carbon/proc/has_disease(var/datum/disease/virus)
+	for(var/datum/disease/D in viruses)
+		if(D.IsSame(virus))
+			return TRUE
+	return FALSE
+
+
+/mob/living/carbon/proc/contract_disease(datum/disease/virus, skip_this = FALSE, force_species_check = TRUE, spread_type = -5)
+	if(species.species_flags & IS_SYNTHETIC)
+		return //synthetic species are immune
+	if(stat == DEAD)
+		//to_chat(world, "He's dead jim.")
+		return
+	if(istype(virus, /datum/disease/advance))
+		//to_chat(world, "It's an advance virus.")
+		var/datum/disease/advance/A = virus
+		if(A.GetDiseaseID() in resistances)
+			//to_chat(world, "It resisted us!")
+			return
+		if(count_by_type(viruses, /datum/disease/advance) >= 3)
+			return
+
+	else
+		if(src.resistances.Find(virus.type))
+			//to_chat(world, "Normal virus and resisted")
+			return
+
+	if(has_disease(virus))
+		return
+
+	if(force_species_check)
+		var/fail = 1
+		if(ishuman(src))
+			var/mob/living/carbon/human/H = src
+			for(var/vuln_species in virus.affected_species)
+				if(H.species.name == vuln_species)
+					fail = 0
+					break
+		if(fail) return
+
+	if(skip_this == 1)
```

Comment:
A lot of lines below that also need cleanup

Reply:
This is all stuff to be removed. Only not done on this PR not to bloat it. Merely moved it.